taipy/core/_orchestrator/_orchestrator.py

```diff
@@ -194,7 +193,7 @@ def __check_if_timeout(st, to):
             return True
 
         start = datetime.now()
-        jobs = jobs if isinstance(jobs, Iterable) else [jobs]
+        jobs = list(jobs) if isinstance(jobs, Iterable) else [jobs]
```

Comment:
💬 Very opinionated suggestion (not blocking, you can just ignore 😄)

What if we change the signature of the class method to `def _wait_until_job_finished(cls, *jobs: Job, timeout: Optional[float] = None)`?
* For the `timeout`, the type `int` is implied in `float` so it's actually not necessary to use an `Union` (see https://peps.python.org/pep-0484/#the-numeric-tower).
* For the `job`, the variadic argument make it more "straightforward" to manipulate as it will always be tuple... No need to use `isinstance` anylonger 😉 

As a bonus: I think that new signature would ne compatible with the old one...

Reply:
you're right! thanks! I'll make the changes in a separate pr as it's minimal :D